src/olympia/amo/tests/test_url_prefix.py

```diff
@@ -38,6 +38,11 @@ def test_redirection(self):
             '/es-PE/firefox/addon/1': '/es/firefox/addon/1',
             # /developers doesn't get an app.
             '/developers': '/en-US/developers',
+            # Shorter locales redirects to long form
```

Comment:
Curious how it would handle invalid urls like 

`/-pt/developers`

or 

`/pt--PT/developers`

or

`/-/developers`

Could you add some of those tests to verify sane handling of weird edge cases?

Reply:
addons-server handles all of these poorly (adding `/en-US/firefox/` prefix to all )... but adding a test would be confusing, because none of them match the regexp set in nginx to redirect traffic to addons-server, so they would all be handled by addons-frontend, which _does_ handle them somewhat correctly, redirecting to either `/en-US/developers/` or `/pt-PT/developers/`

Tbh we don't have to care at all - those are not supposed to work anyway.

Reply:
Oh man so we have 3 different sources of redirection? Fun.